lib/actions/addPrice.js

```diff
@@ -29,17 +29,28 @@ function action(msg, cfg) {
 
     function addPrice(response) {
         var product = response.body;
+        var stagedProduct = product.masterData.staged;
+        var variantId = parseInt(data.variantId, 10);
+
+        var variantObject;
+        if (variantId === 1) {
+            variantObject = stagedProduct.masterVariant;
+        } else {
+            variantObject = _.where(stagedProduct.variants, {id: variantId});
+        }
+
+        var foundPriceInSameScope = variantObject && _.where(variantObject.prices, {currencyCode: data.currency});
```

Comment:
It also might make sense to update the title of the action in ``component.json`` to **Add/Update Variant Price**